python/lsst/validate/drp/validate.py

```diff
@@ -0,0 +1,368 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, absolute_import
+
+import numpy as np
+import yaml
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+
+from .calcSrd import calcAM1, calcAM2
+from .check import checkAstrometry, checkPhotometry, positionRms
+from .print import printPA1, printPA2, printAM1, printAM2
+from .plot import plotAstrometry, plotPhotometry, plotPA1, plotAM1, plotAM2
+from .util import getCcdKeyName, repoNameToPrefix
+
+
+def loadDataIdsAndParameters(configFile):
+    """Load data IDs, magnitude range, and expected metrics from a yaml file.
+
+    Inputs
+    ------
+    configFile : str
+        YAML file that stores visit, filter, ccd,
+        good_mag_limit, medianAstromscatterRef, medianPhotoscatterRef, matchRef
+
+    Returns
+    -------
+    dict, float, float, float
+        dataIds, good_mag_limit, medianRef, matchRef
+    """
+    stream = open(configFile, mode='r')
+    data = yaml.load(stream)
+
+    ccdKeyName = getCcdKeyName(data)
+    visitDataIds = constructDataIds(data['visits'], data['filter'],
+                                    data[ccdKeyName], ccdKeyName)
+
+    return (visitDataIds,
+            data['good_mag_limit'],
+            data['medianAstromscatterRef'],
+            data['medianPhotoscatterRef'],
+            data['matchRef'],
+           )
+
+
+def constructDataIds(visits, filter, ccds, ccdKeyName=None):
+    """Construct a comprehensive set of Butler Data IDs for a repository.
+
+    Inputs
+    ------
+    visits : list of int
+    filter : str
+    ccds : list of int
+    ccdKeyName : str, optional; default 'ccd'
+        Name to distinguish different parts of a focal plane.
+        Generally 'ccd', but might be 'ccdnum', or 'amp', or 'ccdamp'.
+        Refer to your `obs_*/policy/*Mapper.paf`.
+
+    Returns
+    -------
+    list
+        dataIDs suitable to be used with the LSST Butler.
+
+    Examples
+    --------
+    >>> dataIds = constructDataIds([100, 200], 'r', [10, 11, 12])
+    >>> print(dataIds)
+    [{'filter': 'r', 'visit': 100, None: 10}, {'filter': 'r', 'visit': 100, None: 11}, {'filter': 'r', 'visit': 100, None: 12}, {'filter': 'r', 'visit': 200, None: 10}, {'filter': 'r', 'visit': 200, None: 11}, {'filter': 'r', 'visit': 200, None: 12}]
```

Comment:
Ah, bad MWV!  I had copied and pasted the doctest because I couldn't figure what I was getting wrong in the formatting.

It wasn't the formatting.